docs/core/extensions/logging.md

```diff
@@ -3,34 +3,106 @@ title: Logging in C#
 
 [!INCLUDE [logging-samples-browser](includes/logging-samples-browser.md)]
 
-> [!IMPORTANT]
-> Starting with .NET 6, logging services no longer register the <xref:Microsoft.Extensions.Logging.ILogger> type. When using a logger, specify the generic-type alternative <xref:Microsoft.Extensions.Logging.ILogger%601> or register the `ILogger` with [dependency injection (DI)](dependency-injection.md).
+## Logging in a non-trivial app
+
+There are several changes you should consider making to the example above when logging in a less trivial scenario:
+
+- If your application is using [Dependency Injection (DI)](dependency-injection.md) or a host such as ASP.NET's [WebApplication](/aspnet/core/fundamentals/minimal-apis/webapplication) or [Generic Host](generic-host.md) then
+you should use ILoggerFactory and ILogger objects from the DI container rather than creating them directly. See [Integration with DI and Hosts](#integration-with-hosts-and-dependency-injection) below for details.
+
+- Logging [compile-time source generation](logger-message-generator.md) is usually a better alternative to ILogger extension methods like LogInformation(). It offers
+better performance, stronger typing, and avoids spreading string constants throughout your methods. The tradeoff is that using this technique requires a bit more code.
+
+:::code language="csharp" source="snippets/logging/getting-started-logger-message/Program.cs" range="5-13" highlight="5,8-9":::
+
+- The recommended practice for log category names is to use the name of the class that is creating the log message. This helps relate log messages back to the code which produced
+them and offers a good level of control when filtering logs. <xref:Microsoft.Extensions.Logging.LoggerFactoryExtensions.CreateLogger%2A> accepts a Type to make this naming easy to do.
+
+:::code language="csharp" source="snippets/logging/getting-started-type-category-name/Program.cs" range="3-11" highlight="6":::
+
+- If you don't use console logs as your sole production monitoring solution, add the [logging providers](logging-providers.md) you plan to use. For example you could use
+[OpenTelemetry](https://github.com/open-telemetry/opentelemetry-dotnet#getting-started) sending logs over the [OTLP protocol](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md#enable-log-exporter):
+
+:::code language="csharp" source="snippets/logging/getting-started-open-telemetry/Program.cs" range="4-12" highlight="3-6":::
+
+## Integration with hosts and dependency injection
+
+If your application is using [Dependency Injection (DI)](dependency-injection.md) or a host such as ASP.NET's [WebApplication](/aspnet/core/fundamentals/minimal-apis/webapplication) or [Generic Host](generic-host.md) then
+you should use ILoggerFactory and ILogger objects from the DI container rather than creating them directly.
```

Comment:
```suggestion
If your application is using [Dependency Injection (DI)](dependency-injection.md) or a host such as ASP.NET's [WebApplication](/aspnet/core/fundamentals/minimal-apis/webapplication) or [Generic Host](generic-host.md) then you should use `ILoggerFactory` and `ILogger` objects from the DI container rather than creating them directly.
```